parquet-common/src/main/java/org/apache/parquet/io/SeekableInputStream.java

```diff
@@ -105,4 +107,21 @@ public abstract class SeekableInputStream extends InputStream {
    */
   public abstract void readFully(ByteBuffer buf) throws IOException;
 
+  /**
+   * Read a set of file ranges in a vectored manner.
+   * @throws UnsupportedOperationException if not available in this class/runtime.
+   */
+  public void readVectored(List<ParquetFileRange> ranges,
+    IntFunction<ByteBuffer> allocate) throws IOException {
+
+    throw new UnsupportedOperationException("Vectored IO is not supported for " + this);
```

Comment:
Do you have any idea what 'this' will print out?  Like to know if it is useful to user to troubleshot in case of exception